### ISSUE TITLE
The Snackbar component dismiss link needs relevant HTML attributes for keyboard only users

### ISSUE THREAD

**Issue by nicklemmon** [SparkPost/matchbox#196]
The `<Snackbar/>` dismiss button requires several tweaks, as it is not currently keyboard focusable and interactive.

## Requirements

* Dismiss buttons within `<Snackbar/>` component instances should be keyboard focusable
* Dismiss buttons within a component instance need to be interactive using the enter key and spacebar

## Potential Fixes

* Update the HTML to use the `<button>` element instead of the `<a>`
* __or__ use the existing `<a>` element and add `role="button"` alongside a relevant value for `href` - `href="javascript:void(0):"` should do the trick
